lib/ansible/playbook/role/__init__.py

```diff
@@ -372,7 +372,8 @@ def _create_validation_task(self, argument_spec, entrypoint_name):
                 },
             },
             'name': task_name,
-            'tags': ['always'],
+            # Unless role is specifically tagged, the tag is set to 'always'
+            'tags': ['always'] if not self.tags else self.tags,
```

Comment:
I think instead of explicitly adding the tags, we should allow inheritance to work like is done with the `gather_facts` task for plays:

https://github.com/ansible/ansible/blob/e07b4edc547e2a5bd429d1027c0102235616db6c/lib/ansible/executor/play_iterator.py#L164-L166

This would require setting the task ds to a var first, instead of direct returning the dict.